### PR TITLE
Reference update to clean up the Other page

### PR DESCRIPTION
**Pull request by sima-zhu**
This is initial draft. 
Discussion put it here:
https://github.com/Azure/azure-sdk-for-java/issues/15904

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -796,15 +787,6 @@
                 "excludePath": [],
                 "excludepackages": "com.azure.messaging.servicebus.implementation"
             },
-            {
```
Comment (scbedd) [inline package.json:799]: I see that you're removing these from CI. Are we removing them entirely?

File: preview/docs-ref-mapping/reference-preview.yml [hunk 2]
```diff
@@ -70,6 +70,26 @@
       href: ~/docs-ref-services/core-experimental-readme-pre.md
       children:
       - com.azure.core.experimental*
+    - name: Gson Serializer 
+      uid: azure.java.sdk.landingpage.services.sdk.core.GsonSerializer
```
Comment (scbedd) [inline preview/docs-ref-mapping/reference-preview.yml:74]: Updating preview manually can be finicky. Given that UIDs here don't have `.pre` on them, I'm betting you did this manually yes? That's totally ok, but there are two possible easier solutions.

1. Only update `reference-unified`. Next CI run will update `reference-preview` for you.
2. Update `reference-unified`, then locally invoke the script that CI uses to update the the ToCs. That'll also modify it for you!

I can walk you through either one! Gotchas, etc

File: package.json (hunk 1 above)
Comment (sima-zhu) [reply package.json:799]: I am not quite sure about this. I proposed to remove it in the issue listed in description. Java team does not review it yet. These packages does not in the mono repo at all. I am thinking to remove them from doc as well

File: docs-ref-mapping/reference-unified.yml [hunk 3]
```diff
@@ -356,11 +380,17 @@
         landingPageType: Service
         children:
         - com.microsoft.azure.cosmosdb*
+    - name: Spring Data
+      uid: azure.java.sdk.landingpage.services.cosmosdb.SpringData.pre
```
Comment (scbedd) [inline docs-ref-mapping/reference-unified.yml:384]: no `.pre` in unified. just SpringData is fine.

File: docs-ref-mapping/reference-unified.yml [hunk 4]
```diff
@@ -356,11 +380,17 @@
         landingPageType: Service
         children:
         - com.microsoft.azure.cosmosdb*
+    - name: Spring Data
```
Comment (scbedd) [inline docs-ref-mapping/reference-unified.yml:383]: This should be placed with the other `Spring` stuff.

https://github.com/Azure/azure-docs-sdk-java/blob/fc56192ae5ea8c8f82afbe96d4e52310d20c47b0/docs-ref-mapping/reference-unified.yml#L812

See where I mean? I don't think it goes under `cosmos`. You may have details I don't though!

File: package.json (hunk 1 above)
Comment (scbedd) [reply package.json:799]: >These packages do not in the mono repo at all.  

what do you mean by this? Are you saying that this documentation isn't current in docs? I disagree.

image

https://review.docs.microsoft.com/en-us/java/api/overview/azure/core?view=azure-java-preview&branch=master

Comment (scbedd) [reply package.json:799]: Make certain you check the `preview` moniker!

Comment (scbedd) [reply package.json:799]: Resolved after discussion offline. `tracing-opentelemetry` is retired, as is `tracing-opencensus`.

File: docs-ref-mapping/reference-unified.yml (hunk 4 above)
Comment (sima-zhu) [reply docs-ref-mapping/reference-unified.yml:383]: I did not find the best place to go. I found these packages are under sdk/cosmos, so I decided to put it here. The Spring boot and Spring cloud seem not the best place to go. I don't want to give it a new page. 

Let put it here for now. We can simply move to the right position if people complains.

Comment (sima-zhu) [reply docs-ref-mapping/reference-unified.yml:383]: https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/cosmos

Comment (scbedd) [reply docs-ref-mapping/reference-unified.yml:384]: Hmm. Alrighty. Definitely not blocking this PR on this piece of feedback.

**Comment by scbedd** [comment]
I don't believe you called the script properly. Your `preview` is populated with WAY more namespaces than it actually has.  Not only that, it populated in the wrong folder. Did you pull the appropriate commands from the CI job?

**Review by scbedd — CHANGES_REQUESTED** [review]
reference-preview = new file instead of updating existing.

**Comment by sima-zhu** [comment]
Put a wrong path parameter for preview. Regenerate again

**Review by scbedd — APPROVED** [review]
I think this looks good. Worth a shot!

**Pull request merged**